src/engine/strat_engine/thinpooldev.rs

```diff
@@ -0,0 +1,88 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+use devicemapper::{DM, DevId, DeviceInfo, DmFlags};
+use engine::EngineResult;
+
+use std::path::Path;
+
+use super::blockdev::BlockDev;
+
+use types::DataBlocks;
+use types::Sectors;
+
+pub struct ThinPoolDev {
+    pub name: String,
+    dev_info: Option<DeviceInfo>,
+    data_block_size: Option<Sectors>,
+    pub low_water_mark: Option<DataBlocks>,
+    pub meta_dev: Option<BlockDev>,
+    pub data_dev: Option<BlockDev>,
+}
+
+/// support use of DM to create pools for thin provisioned devices
+impl ThinPoolDev {
+    pub fn new(name: &str) -> ThinPoolDev {
+        ThinPoolDev {
+            name: name.to_owned(),
+            dev_info: None,
+            data_block_size: None,
+            low_water_mark: None,
+            meta_dev: None,
+            data_dev: None,
+        }
+    }
+    /// Generate a Vec<> to be passed to DM.  The format of the Vec entries is:
+    /// <start sec> <length> "thin-pool" /dev/meta /dev/data <block size> <low water mark>
+    fn dm_table(&self,
+                length: Sectors,
+                data_block_size: Sectors,
+                low_water_mark: DataBlocks,
+                meta_dev: &Path,
+                data_dev: &Path)
+                -> Vec<(u64, u64, String, String)> {
+        let mut table = Vec::new();
+        let params = format!("{} {} {} {} 1 skip_block_zeroing",
+                             meta_dev.to_string_lossy(),
+                             data_dev.to_string_lossy(),
+                             *data_block_size,
+                             *low_water_mark);
+        table.push((0u64, length.0, "thin-pool".to_owned(), params));
+        debug!("dmtable line : {:?}", table);
```

Comment:
It's nice to have the ability to write out debug info...but I feel we shouldn't put it in in a miscellaneous fashion. It really needs judgement, otherwise you end up with horribly repetitive, inscrutable logs that only experts can read. Also, we might want to strategize about putting information in the journal, in a useful way.

IOW, is it Ok to remove the calls to the ```debug!``` macro now?